extensions/amp-experiment/1.0/variant.js

```diff
@@ -42,12 +41,12 @@ export class Variants {
   }
 
   /**
-   * @param {!Object<string, ?string>|!Promise<!Object<string, ?string>>} variants
+   * @param {!Promise<!Object<string, ?string>>} variants
    * @package
    * @restricted
    */
   init(variants) {
-    this.variantsDeferred_.resolve(variants);
+    variants.then(this.variantsDeferred_.resolve);
```

Comment:
I believe you want to do 
`variants.then(result => this.viriantsDeferred_.resolve(result))` ?